plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -0,0 +1,176 @@
+package io.jenkins.plugins.casc;
+
+import io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class SchemaGeneration {
+
+    final static JSONObject schemaTemplateObject = new JSONObject()
+        .put("$schema", "http://json-schema.org/draft-07/schema#")
+        .put("id", "http://jenkins.io/configuration-as-code#")
+        .put("description", "Jenkins Configuration as Code")
+        .put("type", "object");
```

Comment:
You can also have too much whitespace cleanup
```suggestion
        .put("type", "object");

```

Reply:
@casz Is this perfect?